File: src/main/java/no/novari/flyt/configuration/mapping/ValueMappingMappingService.java
package no.novari.flyt.configuration.mapping;

import no.novari.flyt.configuration.model.configuration.dtos.ValueMappingDto;
import no.novari.flyt.configuration.model.configuration.entities.ValueMapping;
import org.springframework.stereotype.Service;

@Service
public class ValueMappingMappingService {

    public ValueMapping toEntity(ValueMappingDto valueMappingDto) {
        return ValueMapping
                .builder()
                .type(valueMappingDto.getType())
                .mappingString(valueMappingDto.getMappingString())
                .build();
    }

    public ValueMappingDto toDto(ValueMapping valueMapping) {
        return ValueMappingDto
                .builder()
                .type(valueMapping.getType())
                .mappingString(valueMapping.getMappingString())
                .build();
    }

}

File: src/main/kotlin/no/novari/flyt/configuration/mapping/ValueMappingMappingService.kt
package no.novari.flyt.configuration.mapping

import no.novari.flyt.configuration.model.configuration.dtos.ValueMappingDto
import no.novari.flyt.configuration.model.configuration.entities.ValueMapping
import org.springframework.stereotype.Service

@Service
class ValueMappingMappingService {
    fun toEntity(valueMappingDto: ValueMappingDto): ValueMapping =
        ValueMapping
            .builder()
            .type(valueMappingDto.type)
            .mappingString(valueMappingDto.mappingString)
            .build()

    fun toDto(valueMapping: ValueMapping): ValueMappingDto =
        ValueMappingDto
            .builder()
            .type(valueMapping.type)
            .mappingString(valueMapping.mappingString)
            .build()
}
